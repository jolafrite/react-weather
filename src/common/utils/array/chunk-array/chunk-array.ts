export const chunkArray = <T>(array: T[], size: number): T[][] => {
  return array.length >size
    ? [array.slice(0, size), ...chunkArray(array.slice(size), size)]
    : [array];
}
