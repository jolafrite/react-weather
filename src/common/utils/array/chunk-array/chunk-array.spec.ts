import { chunkArray } from "./chunk-array";

describe('chunkArray', () => {

  test('array length lower than size', () => {
    const array = Array.from(Array(6).keys());

    expect(
      chunkArray(array, 6)
    ).toMatchSnapshot();
  });

  test('array can be split evenly', () => {
    const array = Array.from(Array(6).keys());

    expect(
      chunkArray(array, 3)
    ).toMatchSnapshot();
  });

  test('array cant be split evenly', () => {
    const array = Array.from(Array(6).keys());

    expect(
      chunkArray(array, 4)
    ).toMatchSnapshot();
  });

});
