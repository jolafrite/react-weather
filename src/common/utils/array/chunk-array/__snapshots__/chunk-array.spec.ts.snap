// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chunkArray array can be split evenly 1`] = `
Array [
  Array [
    0,
    1,
    2,
  ],
  Array [
    3,
    4,
    5,
  ],
]
`;

exports[`chunkArray array cant be split evenly 1`] = `
Array [
  Array [
    0,
    1,
    2,
    3,
  ],
  Array [
    4,
    5,
  ],
]
`;

exports[`chunkArray array length lower than size 1`] = `
Array [
  Array [
    0,
    1,
    2,
    3,
    4,
    5,
  ],
]
`;
